### PR TITLE
Initialize the scenario controller by kubebuilder

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Let's start our journey :)

This PR just initializes the scenario controller.

What I did is:
- `kubebuilder init --domain kube-scheduler-simulator.x-k8s.io  --repo sigs.k8s.io/kube-scheduler-simulator/scenario`
- `kubebuilder create api --group simulation --version v1alpha1 --kind Scenario`
- edit readme.md

#### Which issue(s) this PR fixes:

Part of #140

#### Special notes for your reviewer:

/cc @everpeace 
/cc @196Ikuchil 

/label tide/merge-method-squash

### CODE REVIEW

File: scenario/go.mod
Comment (sanposhiho) [inline scenario/go.mod:1]: This PR creates it as another module. You may want to discuss it.

My point is:
- by creating it as a separate module, we can completely separate the logic. It may help us in the future like when we want to expand it to be run in user's cluster. 
  - And of course, it makes it easier to maintain. (those who want to develop this controller don't need to know about simulator's implementation.)
- we can run the controller in another container of the simulator's Pod. 
  - docker-compose as well.

Comment (sanposhiho) [reply scenario/go.mod:1]: > we can run the controller in another container of the simulator's Pod

It requires this change on KEP: https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/194

Comment (everpeace) [reply scenario/go.mod]: Yeah, upon #194 getting merged, I agree that we make the controller the new module.
I ack-ed https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/196/commits/24a77a823391896ed1a8c340844e6f1b70b5c9ec changed the module name.

**Review by everpeace — APPROVED** [review]
The change is LGTM. How about adding CI??

**Comment by sanposhiho** [comment]
> How about adding CI??

Yeah, definitely, we need to add that. I'll work on that in another PR. :=1

**Comment by sanposhiho** [comment]
@everpeace 
I think you can do `/lgtm` for this PR. 
For #194 I'd like to wait for reviews from @196Ikuchil since the KEP reviewer is him. But this PR only does implementation and a review from one person is enough like other PRs.
---
To wait for #194.
/hold

File: scenario/go.mod
Comment (196Ikuchil) [inline scenario/go.mod:3]: Is `go1.18` ok?
Our `simulator` `go.mod` specifies `go1.17`.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/bd9851654b19a1a123d44075249814d14193faa2/simulator/go.mod#L3

Or, is this `scenario` completely isolated project from the simulator?

Comment (sanposhiho) [reply scenario/go.mod]: It's an isolated go module so I think either is fine. 
But, let me change it to 1.17 to be same as other go module 念の為 :)

**Review by 196Ikuchil — COMMENTED** [review]
Nits. 
Created in kubebuilder?
I'll leave it up to you to fix it.

**Comment by sanposhiho** [comment]
/unhold

https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/194 now is a quite big change. 
I think we can start the implementation since some designs that have already been agreed upon can already be implemented.

**Comment by sanposhiho** [comment]
@196Ikuchil Thanks. Fixed.

**Comment by 196Ikuchil** [comment]
> I think we can start the implementation since some designs that have already been agreed upon can already be implemented.

ok, agreed!

/lgtm

**Pull request merged**
